src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -54,7 +54,7 @@ class RsPsiFactory(private val project: Project) {
         createExpressionOfType("unsafe { $body }")
 
     fun tryCreatePath(text: String): RsPath? {
-        val path = createFromText<RsPathExpr>("fn main() { $text;}")?.path ?: return null
+        val path = createFromText<RsPathExpr>("const a = $text;")?.path ?: return null
```

Comment:
Since path expr resolution now delegated to type inference, `fn main() { $text;}` will perform type inference first, sometimes during type inference of another fn. Not only is it forbidden, it's also just useless work. `const a = $text;` is a very temporary solution because const exprs really should call type inference too. But there is no other syntax for **value** path including paths to associated items. Good news is that path to associated item (like `<A as B>::C`) used is tests only (khm, really in just a single test), so I think here we can use just `use $text;` and leave plain old `fn main() { $text;}` for test cases.